lib/bolt_server/transport_app.rb

```diff
@@ -366,21 +370,30 @@ def plan_list(pal)
       plans.map { |plan_name| { 'name' => plan_name } }
     end
 
-    def file_metadatas(pal, module_name, file)
-      pal.in_bolt_compiler do
-        mod = Puppet.lookup(:current_environment).module(module_name)
-        raise ArgumentError, "`module_name`: #{module_name} does not exist" unless mod
-        abs_file_path = mod.file(file)
-        raise ArgumentError, "`file`: #{file} does not exist inside the module's 'files' directory" unless abs_file_path
-        fileset = Puppet::FileServing::Fileset.new(abs_file_path, 'recurse' => 'yes')
-        Puppet::FileServing::Fileset.merge(fileset).collect do |relative_file_path, base_path|
-          metadata = Puppet::FileServing::Metadata.new(base_path, relative_path: relative_file_path)
-          metadata.checksum_type = 'sha256'
-          metadata.links = 'follow'
-          metadata.collect
-          metadata.to_data_hash
+    def file_metadatas(versioned_project, module_name, file)
+      abs_file_path = nil
+      @pal_mutex.synchronize do
+        bolt_config = config_from_project(versioned_project)
+        pal = pal_from_project_bolt_config(bolt_config)
+        pal.in_bolt_compiler do
+          mod = Puppet.lookup(:current_environment).module(module_name)
+          raise ArgumentError, "`module_name`: #{module_name} does not exist" unless mod
+          abs_file_path = mod.file(file)
+          unless abs_file_path
+            raise ArgumentError,
+                  "`file`: #{file} does not exist inside the module's 'files' directory"
+          end
         end
       end
```

Comment:
```suggestion
    def file_metadatas(versioned_project, module_name, file)
      abs_file_path = @pal_mutex.synchronize do
        bolt_config = config_from_project(versioned_project)
        pal = pal_from_project_bolt_config(bolt_config)
        pal.in_bolt_compiler do
          mod = Puppet.lookup(:current_environment).module(module_name)
          raise ArgumentError, "`module_name`: #{module_name} does not exist" unless mod
          abs_file_path = mod.file(file)
          unless abs_file_path
            raise ArgumentError,
                  "`file`: #{file} does not exist inside the module's 'files' directory"
          end
          abs_file_path
        end
      end
```

Is there a way to assign `abs_file_path` from the result of the block instead of initializing to nil and mutating it in the block?

Reply:
Sure, done.